src/containers/color_generator/index.js

```diff
@@ -14,25 +14,29 @@ export class ColorGenerator extends Component {
   }
 
   componentDidMount() {
-    this.generateColors()
+    const {colors} = this.state; 
+    this.generateColors(colors)
   }
 
-  bringInColors = () => {
+  componentDidUpdate(prevProps, prevState) {
     const {currentPalette} = this.props;
-    let colors = [];
-    for (let i = 1; i <= 5; i++) {
-      let currColor = {
-        hex: currentPalette[`color_${i}`],
-        locked: false
-      };
-      colors.push(currColor)
+    const currentPaletteHasChanged = currentPalette.id && prevProps.currentPalette.id !== currentPalette.id
```

Comment:
I assume you got this working 😱 That's GREAT NEWS!!
- I plan on using this to hopefully get the inputs working